notebooks/setfit-absa-fiqa.ipynb

```diff
@@ -0,0 +1,7780 @@
+{
```

Comment:
<p>In their paper [BloombergGPT: A Large Language Model for Finance](<a href="https://arxiv.org/pdf/2303.17564.pdf)" target="_blank">https://arxiv.org/pdf/2303.17564.pdf)</a>, Bloomberg presents the development of <strong>**BloombergGPT**</strong>, LLM which is specialized for the financial domain.&lt;br&gt;</p><p>The model is a <strong>**50B**</strong> parameters language model, trained on a wide range of financial data:</p><p> </p><p>* &nbsp; <strong>**363 billion**</strong> tokens dataset based on Bloomberg's extensive data sources</p><p>* &nbsp; <strong>**345 billion**</strong> tokens from general purpose datasets</p><p> </p><p>The model was evaluated on various financial NLP tasks, including Aspect-Based Sentiment Analysis (<strong>**ABSA**</strong>). Bloomberg evaluated ABSA in an in-context manner using the [FiQA_SA](<a href="https://huggingface.co/datasets/AdaptLLM/finance-tasks/viewer/FiQA_SA)" target="_blank">https://huggingface.co/datasets/AdaptLLM/finance-tasks/viewer/FiQA_SA)</a>. This dataset contains a test set of 235 finance related sentences, each is prefixed by 5 tagged sentences, which overall comprise an input prompt to the model. Every tagged sentence in the prompt contains a known aspect and a question whether its corresponding polarity is Positive, Negative or Neutral, and the answer (the correct polarity towards the mentioned aspect) is written at the end. BloombergGPT is expected to predict the polarity of a given aspect in the test sentence based on the 5 tagged example sentences. This is an <strong>**SB2 ABSA task**</strong>, which means that the aspect is already given and the model just needs to predict the corresponding polarity.</p><p> </p><p>The evaluation score achieved by this model is <strong>**weighted F1-score=75.07**</strong>, as reported in table 8 in the paper.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/huggingface/setfit/pull/471/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>